### PR TITLE
Remove oudated subscription update logic to improve resolution delay

### PR DESCRIPTION
**Pull request by dinhxuanvu**
Currently, olm logic checks for upgrade in subscription via another
obsolete API that is no longer in use for dependency solution. As a
result, sometimes, subscriptions display `UpgradeAvailable` status but
there will be no upgrades as the upgrade is not valid in the resolver.
Also, the `UpgradeAvailable` status is used to trigger the new resolution
even though that status is no longer a valid indicator of having a pending
upgrade. This leads to unwanted upgrade delay when the obsolete API works
properly.

This commit will remove the code that is using this obsolete API and
allow the resolution to happen when there is a subscription change.

Signed-off-by: Vu Dinh <vudinh@outlook.com>

**Description of the change:**

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

**Comment by dinhxuanvu** [comment]
@kevinrizza Need a re-lgtm if you don't mind. Thanks.

File: pkg/controller/operators/catalog/operator.go [hunk 1]
```diff
@@ -1112,14 +1107,7 @@ func (o *Operator) ensureSubscriptionCSVState(logger *logrus.Entry, sub *v1alpha
 		if err := querier.Queryable(); err != nil {
 			return nil, false, err
 		}
```
Comment (benluddy) [inline pkg/controller/operators/catalog/operator.go:1109]: This can be removed, too, right?

**Comment by nunnatsa** [comment]
@dinhxuanvu - HCO upgrade tests start failing in prow after merging this PR. The test subscribes to a previous version and then performs an upgrade to the PR version (the version under test). Could you please check?

**Comment by dinhxuanvu** [comment]
@nunnatsa Can you create a thread on Slack and tag me on it? I need some further logging and info to identify what the issue may be.

**Comment by nunnatsa** [comment]
> @nunnatsa Can you create a thread on Slack and tag me on it? I need some further logging and info to identify what the issue may be.

There are no relevant logs from OLM, because OLM just ignores the change in the subscription.

**Pull request merged**